### PR TITLE
Gracefully handle null data in refresh offer message and log error

### LINKED ISSUE #5786: Cannot create P2PDataStorage.ByteArray with empty byte[] array argument.
Woke up this morning to this error.   
[bisq.log](https://github.com/bisq-network/bisq/files/7441642/bisq.log)

### PR DESCRIPTION
**Pull request by ghost**
User reported an unhandled exception popup in v1.7.5.

Just before the issue the client did get a disconnect: `b.n.p.n.Connection: proto is null because protoInputStream.read()=-1 (EOF). That is expected if client got stopped without proper shutdown.`

A offer refresh TTL message should not contain incomplete data, but is possible if the network connection is interrupted.  There are other examples of interrupted network messages already logged.

Fix is to catch and log the exception stack trace.  Just adds a try..catch block in `P2PDataStorage.refreshTTL`.
Fixes #5786

_If there is an alternative way to handle this, let me know.  @bisq-network/bisq-devs_ 

---

### Before

crash_p2p_exception

### After:

**[[NO POPUP SHOWN]]**

**and logfile contains a stack trace of the error:**

    WARN  b.n.p.n.Connection: proto is null because protoInputStream.read()=-1 (EOF). That is expected if client got stopped without proper shutdown. 
    java.lang.IllegalArgumentException: Cannot create P2PDataStorage.ByteArray with null byte[] array argument.
	    at bisq.network.p2p.storage.P2PDataStorage$ByteArray.verifyBytesNotEmpty(P2PDataStorage.java:1193)
	    at bisq.network.p2p.storage.P2PDataStorage$ByteArray.<init>(P2PDataStorage.java:1188)
	    at bisq.network.p2p.storage.P2PDataStorage.refreshTTL(P2PDataStorage.java:850)
	    at bisq.network.p2p.storage.P2PDataStorage.lambda$onMessage$22(P2PDataStorage.java:585)
	    at java.base/java.util.Optional.ifPresent(Optional.java:183)
	    at bisq.network.p2p.storage.P2PDataStorage.onMessage(P2PDataStorage.java:577)
	    at bisq.network.p2p.network.NetworkNode.lambda$onMessage$6(NetworkNode.java:392)
	    at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	    at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	    at bisq.network.p2p.network.NetworkNode.onMessage(NetworkNode.java:392)
	    at bisq.network.p2p.network.Connection.lambda$onBundleOfEnvelopes$11(Connection.java:470)
	    at java.base/java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:803)
	    at java.base/java.util.concurrent.CopyOnWriteArraySet.forEach(CopyOnWriteArraySet.java:425)
	    at bisq.network.p2p.network.Connection.lambda$onBundleOfEnvelopes$12(Connection.java:470)
	    at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
	    at java.base/java.security.AccessController.doPrivileged(Native Method)
	    at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
	    at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	    at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	    at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	    at java.base/java.lang.Thread.run(Thread.java:834)

### CODE REVIEW

File: p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java [hunk 1]
```diff
@@ -847,42 +847,47 @@ public boolean hasAlreadyRemovedAddOncePayload(ProtectedStoragePayload protected
...
+            sequenceNumberMap.put(hashOfPayload, new MapValue(updatedEntry.getSequenceNumber(), this.clock.millis()));
+            requestPersistence();
 
-        // Always broadcast refreshes
-        broadcaster.broadcast(refreshTTLMessage, sender);
+            // Always broadcast refreshes
+            broadcaster.broadcast(refreshTTLMessage, sender);
 
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
```
Comment (ripcurlx) [inline p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java:889]: Maybe we should add an error log to give the stacktrace more context in the log file?

**Comment by ghost** [comment]
Per request, added error log message.  It looks like this:

`Nov-02 09:28:57.426 [JavaFX Application Thread] ERROR b.n.p2p.storage.P2PDataStorage: refreshTTL failed, missing data: java.lang.IllegalArgumentException: Cannot create P2PDataStorage.ByteArray with null byte[] array argument. `
[+followed by stack trace]

**Review by ripcurlx — APPROVED** [review]
utACK

**Review by chimp1984 — APPROVED** [review]
utACK

**Pull request merged**